File: Leetcode_Java/Integer/PalindromeNumber.java
package Leetcode_Java.Integer;

public class PalindromeNumber {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        int palindrome = 0;
        int dummy = x;

        while (dummy != 0) {
            int curr = dummy % 10;
            dummy /= 10;
            palindrome = palindrome * 10 + curr;
        }
        return palindrome == x;
    }
}

File: Leetcode_Python/Integer/palindrome_number.py
def isPalindrome(self, x):
    """
    :type x: int
    :rtype: bool
    """
    if x < 0:
        return False
    palindrome = 0;
    dummy = x;

    while dummy != 0:
        curr = dummy % 10
        palindrome = palindrome * 10 + curr
        dummy /= 10

    return palindrome == x;